packages/commonwealth/client/scripts/views/pages/CommunityManagement/Topics/CreateTopicsSection/CreateTopicSection.tsx

```diff
@@ -0,0 +1,181 @@
+
+              if (disallowedCharMatches) {
+                const err = `The ${pluralizeWithoutNumberPrefix(
+                  disallowedCharMatches.length,
+                  'char',
+                )}
+              ${disallowedCharMatches.join(', ')} are not permitted`;
+                setErrorMsg(err);
+                return ['failure', err];
+              }
+
+              if (errorMsg) {
+                setErrorMsg(null);
+              }
+
+              return ['success', 'Valid topic name'];
+            }}
+            autoFocus
+          />
+          <CWTextInput
+            label="Description"
+            placeholder="Enter a description"
+            name="description"
+            tabIndex={2}
+            value={description}
+            onInput={(e) => {
+              setDescription(e.target.value);
+            }}
+          />
+          <CWText type="caption">
+            Choose where to feature this topic. Select at least one.
+          </CWText>
+          <CWCheckbox
+            className="sidebar-feature-checkbox"
+            label={
+              <div>
+                <CWText type="b2">
+                  Featured topic in sidebar (recommended)
+                </CWText>
+                <CWText type="caption" className="checkbox-label-caption">
+                  Please note, only sidebar-featured topics show on the Overview
+                  page.
+                </CWText>
+              </div>
+            }
+            checked={featuredInSidebar}
+            onChange={() => {
+              setFeaturedInSidebar(!featuredInSidebar);
+            }}
+            value=""
+          />
+          <CWCheckbox
+            label="Featured in new post"
+            checked={featuredInNewPost}
+            onChange={() => {
+              setFeaturedInNewPost(!featuredInNewPost);
+            }}
+            value=""
+          />
+          {featuredInNewPost && (
+            <ReactQuillEditor
+              contentDelta={contentDelta}
+              setContentDelta={setContentDelta}
+            />
+          )}
+        </div>
+        <div className="actions">
+          <CWButton
+            label="Create topic"
+            buttonType="primary"
+            buttonHeight="med"
+            buttonWidth="wide"
+            disabled={isSaving || !!errorMsg}
+            onClick={async (e: React.MouseEvent<HTMLButtonElement>) => {
+              e.preventDefault();
+              try {
+                await createTopic({
+                  name,
+                  description,
+                  featuredInSidebar,
+                  featuredInNewPost,
+                  defaultOffchainTemplate: serializeDelta(contentDelta),
+                });
+                navigate(`/discussions/${encodeURI(name.toString().trim())}`);
+              } catch (err) {
+                setErrorMsg('Error creating topic');
+                setIsSaving(false);
+              }
+            }}
```

Comment:
move this up to separate function handler in order to declutter the markup

Reply:
Done!